### PR TITLE
feat(sdk): add `cloud.Counter` for `tf-azure` platform

### LINKED ISSUE #629: cloud.Counter - Azure implementation

### PR DESCRIPTION
**Pull request by garysassano**
Closes #629

## Checklist

- [x] Title matches [Winglang's style guide](https://www.winglang.io/contributing/start-here/pull_requests#how-are-pull-request-titles-formatted)
- [x] Description explains motivation and solution
- [ ] Tests added (always)
- [ ] Docs updated (only required for features)
- [ ] Added `pr/e2e-full` label if this feature requires end-to-end testing

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Wing Cloud Contribution License](https://github.com/winglang/wing/blob/main/CONTRIBUTION_LICENSE.md)*.

### CODE REVIEW

File: docs/docs/04-standard-library/helpers/api-reference.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+---
```
Comment (tsuf239) [inline docs/docs/04-standard-library/helpers/api-reference.md:1]: what is this file? can it be removed?

**Review by tsuf239 — COMMENTED** [review]
It looks good, I wonder if you can create a test file including only the `inc` method, so I can test it (first with no initial value, and then with one)
also, if you can add ts test it will be great!

File: docs/docs/04-standard-library/helpers/api-reference.md (hunk 1 above)
Comment (garysassano) [reply docs/docs/04-standard-library/helpers/api-reference.md:1]: I think so since it's automatically generated.

**Comment by tsuf239** [comment]
image
all methods work now :) the only thing missing was the credentials to the table

**Comment by tsuf239** [comment]
@garysassano if you can add ts tests for the rest of the methods it will be great 😻 (if not, I'll take care of it whenever I have more time)
@winglang/maintainers if someone can review my changes it would be great :)  
(I committed straight to the branch since Gary faced some unexplained errors, and we wanted to move forward with that)

File: libs/wingsdk/src/target-tf-azure/counter.ts [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+import { IInflightHost } from "../std";
+
+/**
+ * - Table names must be unique within an account.
+ * - Table names may contain only alphanumeric characters.
+ * - Table names cannot begin with a numeric character.
+ * - Table names are case-insensitive.
+ * - Table names must be from 3 to 63 characters long.
+ * - Some table names are reserved, including "tables".
+ *   Attempting to create a table with a reserved table name returns error code 404 (Bad Request).
```
Comment (hasanaburayyan) [inline libs/wingsdk/src/target-tf-azure/counter.ts:23]: Can we add a link to where these restrictions come from (I assume Azure docs)

File: libs/wingsdk/src/target-tf-azure/counter.ts [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+      cloud.CounterInflightMethods.PEEK,
+    ];
+  }
+
+  public onLift(host: IInflightHost, ops: string[]): void {
+    if (!(host instanceof Function)) {
+      throw new Error("counters can only be bound by tfazure.Function for now");
+    }
+
+    // TODO: investigate customized roles over builtin for finer grained access control
```
Comment (hasanaburayyan) [inline libs/wingsdk/src/target-tf-azure/counter.ts:80]: Tracking issue?

**Review by hasanaburayyan — APPROVED** [review]
Dont know much about Azure, but LGTM add few minor comments

**Review by tsuf239 — APPROVED** [review]
Perfect! Thank you Gary, great job as always 💪 ✨ 🎉

**Review by tsuf239 — APPROVED** [review]

**Pull request merged**